### ISSUE TITLE
Failed to build on Cloudflare Pages - functions were not configured to run with the Edge Runtime: publish.func

### ISSUE THREAD

**Issue by centminmod** [enricoros/nextjs-chatgpt-app#72]
@enricoros updated my fork https://github.com/centminmod/nextjs-chatgpt-app/commit/55490d94400b1e37e11cdf60bd815855f6f5fa35 to latest build and seems I can no longer build the app on Cloudflare Pages? Checking https://nextjs.org/docs/api-routes/edge-api-route gives 404 not found?

Maybe something specific to Vercel? I never used Vercel and built in past straight with Cloudflare Pages though they have https://developers.cloudflare.com/pages/migrations/migrating-from-vercel/

The build errors are related to ` following functions were not configured to run with the Edge Runtime: publish.func`

```
06:36:44.042 | ▲ Detected Next.js version: 13.3.0
-- | --
06:36:44.047 | ▲ Detected `package-lock.json` generated by npm 7+...
06:36:44.048 | ▲ Running "npm run build"
06:36:44.579 | ▲ npm
06:36:44.579 | ▲ WARN
06:36:44.579 | ▲ config tmp This setting is no longer used.  npm stores temporary files in a special
06:36:44.579 | ▲ npm WARN
06:36:44.580 | ▲ config location in the cache, and they are managed by
06:36:44.580 | ▲ npm WARN config     [`cacache`](http://npm.im/cacache).
06:36:44.597 | ▲ > nextjs-chatgpt-app@0.2.0 build
06:36:44.598 | ▲ > next build
06:36:45.367 | ▲ info  - Linting and checking validity of types...
06:36:54.264 | ▲ info  - Creating an optimized production build...
06:37:09.793 | ▲ info  - Compiled successfully
06:37:09.794 | ▲ info  - Collecting page data...
06:37:27.991 | ▲ info  - Generating static pages (0/3)
06:37:29.285 | ▲ info  - Generating static pages (3/3)
06:37:29.306 | ▲ info  - Finalizing page optimization...
06:37:29.314 | ▲
06:37:29.325 | ▲ Route (pages)                              Size     First Load JS
06:37:29.326 | ▲ ┌ ○ /                                      152 kB          251 kB
06:37:29.327 | ▲ ├   └ css/86ad80c02bda2636.css             648 B
06:37:29.327 | ▲ ├   /_app                                  0 B            99.5 kB
06:37:29.327 | ▲ ├ ○ /404                                   182 B          99.7 kB
06:37:29.327 | ▲ ├ ℇ /api/openai/models                     0 B            99.5 kB
06:37:29.327 | ▲ ├ ℇ /api/openai/stream-chat                0 B            99.5 kB
06:37:29.328 | ▲ └ λ /api/publish                           0 B            99.5 kB
06:37:29.328 | ▲ + First Load JS shared by all              105 kB
06:37:29.328 | ▲ ├ chunks/framework-2c79e2a64abdb08b.js   45.2 kB
06:37:29.328 | ▲ ├ chunks/main-4dcb7f9b52833aba.js        27.2 kB
06:37:29.329 | ▲ ├ chunks/pages/_app-02967b3b3aa6bbc9.js  24.7 kB
06:37:29.329 | ▲ ├ chunks/webpack-62a659d736ca0b7b.js     2.44 kB
06:37:29.330 | ▲ └ css/a90f819cc1334814.css               5.27 kB
06:37:29.330 | ▲ ℇ  (Streaming)  server-side renders with streaming (uses React 18 SSR streaming or Server Components)
06:37:29.330 | ▲ λ  (Server)     server-side renders at runtime (uses getInitialProps or getServerSideProps)
06:37:29.330 | ▲ ○  (Static)     automatically rendered as static HTML (uses no initial props)
06:37:34.300 | ▲ Traced Next.js server files in: 4.880s
06:37:35.313 | ▲ Created all serverless functions in: 1.011s
06:37:35.417 | ▲ Collected static files (public/, static/, .next/static): 17.165ms
06:37:37.116 | ▲ Build Completed in .vercel/output [54s]
06:37:37.255 | ⚡️ Building Completed.
06:37:37.441 | ⚡️ ERROR: Failed to produce a Cloudflare Pages build from the project.
06:37:37.441 | ⚡️
06:37:37.441 | ⚡️ The following functions were not configured to run with the Edge Runtime:
06:37:37.441 | ⚡️  - publish.func
06:37:37.441 | ⚡️
06:37:37.442 | ⚡️ If this is a Next.js project:
06:37:37.442 | ⚡️
06:37:37.442 | ⚡️ - you can read more about configuring Edge API Routes here: https://nextjs.org/docs/api-routes/edge-api-route
06:37:37.442 | ⚡️
06:37:37.442 | ⚡️ - you can try enabling the Edge Runtime for a specific page by exporting the following from your page:
06:37:37.442 | ⚡️
06:37:37.442 | ⚡️         export const config = { runtime: 'edge' };
06:37:37.442 | ⚡️
06:37:37.443 | ⚡️ - or you can try enabling the Edge Runtime for all pages in your project by adding the following to your 'next.config.js' file:
06:37:37.443 | ⚡️
06:37:37.443 | ⚡️         const nextConfig = { experimental: { runtime: 'edge'} };
06:37:37.443 | ⚡️
06:37:37.443 | ⚡️ You can read more about the Edge Runtime here: https://nextjs.org/docs/advanced-features/react-18/switchable-runtime
06:37:37.477 | Failed: build command exited with code: 1
06:37:38.454 | Failed: error occurred while running build command
```

The basics of Cloudflare Pages steps for building the app that worked previously was as follows:

For setting up your nextjs app on Cloudflare Pages you can follow Cloudflare developer docs at https://developers.cloudflare.com/pages/framework-guides/deploy-a-nextjs-site/ and https://developers.cloudflare.com/pages/platform/build-configuration/

The only difference is I took these steps below:

1. I forked your repo to my own Github repo
2. On the Cloudflare Pages section, click `Create a project` button > `Connect To Git` and give Cloudflare Pages either All Github account Repo access or selected Repo access. I use selected Repo access and select the forked repo from step 1
3. Once you select the forked Github repo, you click `Begin Setup` button to setup build and deployments. On this page you set you `Project name`, `Production branch` i.e. main and your Build settings where you select from `Framework preset` dropdown menu `Next.js`. Leave the preset filled Build command and Build output directory as preset defaults. You'd want to set `Environmental variables` (advanced) on this page to configure some variables as follows:

VARIABLE |  VALUE
-- | --
GO_VERSION | 1.16
NEXT_TELEMETRY_DISABLED | 1
NODE_VERSION | 17
PHP_VERSION | 7.4
PYTHON_VERSION | 3.7
RUBY_VERSION | 2.7.1

4. Click the Save and Deploy button
5. Then watch the process run to initialize your build environment, clone Github repo, build the application and deploy to Cloudflare Network once that is done, proceed to the project you created.
6. Custom domains tab allows you to set up your domain via CNAME
7. Settings page will have 2 settings you want to enable, Access Policy to restrict [preview deployments](https://developers.cloudflare.com/pages/platform/preview-deployments/) to members of your Cloudflare account via one time pin and to restrict primary `*.YOURPROJECT.pages.dev` domain see https://developers.cloudflare.com/pages/platform/known-issues/#enabling-access-on-your-pagesdev-domain and enable Web Analytics

**Comment by centminmod** [comment]
Not sure what this means but related to https://github.com/cloudflare/next-on-pages#2-configure-the-application-to-use-the-edge-runtime ?

In order for your application to run on Cloudflare Pages, it needs to be set to use the Edge Runtime. Make sure that all the files in your application containing server-side code (e.g. any API Routes and any pages which use getServerSideProps) export a config object specifying the use of the Edge Runtime:

```
export const config = {
	runtime: 'edge',
};

```
Additionally, ensure that your application is not using any [unsupported APIs](https://nextjs.org/docs/api-reference/edge-runtime#unsupported-apis) and that its API routes are defined as [Edge API Routes](https://nextjs.org/docs/api-routes/edge-api-routes).

**Comment by centminmod** [comment]
Dario from Cloudflare has chimed in with a fix and pull request to @enricoros repo at https://github.com/enricoros/nextjs-chatgpt-app/pull/74 :)

**Comment by enricoros** [comment]
Love it. Have to test it and merge it later today. Good fix.

**Comment by centminmod** [comment]
Looks to be working with that pull request fix, can build on Cloudflare Pages again now

![image](https://user-images.githubusercontent.com/5899584/231271305-50680434-a878-4194-a236-a1aae8379f85.png)

**Comment by adwaith182** [comment]
<html>
<body>

⚡️ Completed `npx vercel build`.
--
16:44:41.407 |  
16:44:41.407 | ⚡️ ERROR: Failed to produce a Cloudflare Pages build from the project.
16:44:41.408 | ⚡️
16:44:41.408 | ⚡️ 	The following routes were not configured to run with the Edge Runtime:
16:44:41.408 | ⚡️ 	  - /api
16:44:41.408 | ⚡️
16:44:41.408 | ⚡️ 	Please make sure that all your non-static routes export the following edge runtime route segment config:
16:44:41.408 | ⚡️ 	  export const runtime = 'edge';
16:44:41.408 | ⚡️
16:44:41.409 | ⚡️ 	You can read more about the Edge Runtime on the Next.js documentation:


</body>
</html>

i have the same error ,no fix is helping me 
is there anyone who has solved it 
i did the above fixes also but still not getting fixed

**Comment by enricoros** [comment]
Hi @adwaith182 - please refer to the [docs/deploy-cloudflare.md](https://github.com/enricoros/big-AGI/blob/main/docs/deploy-cloudflare.md) document.
 - in particular, please check if you set the build command to `rm app/api/trpc-node/[trpc]/route.ts && npx @cloudflare/next-on-pages@1`

If you do, please report the new error in a new ticket, not on this one (closed already)

**Comment by codeirawan** [comment]
<html>
<body>

13:26:49.114 | ⚡️ Completed `npx vercel build`.
-- | --
13:26:50.433 |  
13:26:50.433 | ⚡️ ERROR: Failed to produce a Cloudflare Pages build from the project.
13:26:50.434 | ⚡️
13:26:50.434 | ⚡️ 	The following routes were not configured to run with the Edge Runtime:
13:26:50.434 | ⚡️ 	  - /api/corporations
13:26:50.434 | ⚡️ 	  - /api/facilities
13:26:50.434 | ⚡️ 	  - /api/programs
13:26:50.435 | ⚡️ 	  - /api/regions
13:26:50.435 | ⚡️ 	  - /api/search
13:26:50.435 | ⚡️
13:26:50.435 | ⚡️ 	Please make sure that all your non-static routes export the following edge runtime route segment config:
13:26:50.435 | ⚡️ 	  export const runtime = 'edge';
13:26:50.436 | ⚡️
13:26:50.436 | ⚡️ 	You can read more about the Edge Runtime on the Next.js documentation:
13:26:50.436 | ⚡️ 	  https://nextjs.org/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes


</body>
</html>
